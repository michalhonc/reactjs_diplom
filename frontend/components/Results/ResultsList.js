import React from 'react';
import {  } from 'prop-types';
import styled from 'styled-components';

const NON_EXTENDED_ITEM = 5;

const StyledListItem = styled.li`
    list-style: none;
    display: flex;
    justify-content: space-between;
    border-bottom: 2px solid #eee;
    padding: 0.5rem;
`;

const StyledList = styled.ul`
    padding: 0;
    margin: 0;
`;

const StyledButton = styled.button`
    border: 0;
    padding: 0;
    margin: 0;
    background-color: transparent;
    color: #999;
    font-size: 1.4rem;
    line-height: 2rem;
    text-decoration: underline;
    cursor: pointer;
`;

const ResultsList = ({ list }) => {
    const[isExtended, setIsExtended] = React.useState(false);
    const listToRender = isExtended ? list : list.slice(0, NON_EXTENDED_ITEM);
    return (
        <>
            <StyledList>
                {listToRender.map((item, index) => {
                    return (
                        <StyledListItem key={item.id}>
                            <span>{index + 1},</span>
                            <span>{item.full_name}</span>
                            <span>{item.size}tis.</span>
                        </StyledListItem>
                    )
                })}
            </StyledList>
            {!isExtended && (
                    <StyledButton onClick={() => setIsExtended(true)}>
                        Zobrazit další
                    </StyledButton>
                )
            }
        </> 
    )
};

//ResultsList.propTypes = {
//	
//}
//
//ResultsList.defaultProps = {
//	
//}

export default ResultsList;
