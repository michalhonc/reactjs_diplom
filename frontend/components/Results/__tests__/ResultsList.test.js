import React from 'react';
import ResultsList from '../ResultsList';
import renderer from 'react-test-renderer';

it('ResultsList render 5 items', () => {
    const list = [
        {
            id: 'xu19das0',
            full_name: 'Dotaz 1',
            size: 12,
        },
        {
            id: 'xu1ddas0',
            full_name: 'Dotaz 1',
            size: 12,
        },
        {
            id: 'fu19das0',
            full_name: 'Dotaz 1',
            size: 12,
        },
        {
            id: 'x219das0',
            full_name: 'Dotaz 1',
            size: 12,
        },
        {
            id: 'xu193as0',
            full_name: 'Dotaz 1',
            size: 12,
        },
        // ...
    ]
    const tree = renderer.create(
        <ResultsList list={list} />
    );
    expect(tree.toJSON())
        .toMatchSnapshot();
});
