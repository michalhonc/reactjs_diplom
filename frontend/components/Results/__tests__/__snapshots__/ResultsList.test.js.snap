// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList render 5 items 1`] = `
Array [
  <ul
    className="ResultsList__StyledList-sc-1lpwubp-1 eolDkk"
  >
    <li
      className="ResultsList__StyledListItem-sc-1lpwubp-0 blEIsL"
    >
      <span>
        1
        .
      </span>
      <span>
        Dotaz 1
      </span>
      <span>
        12
        tis.
      </span>
    </li>
    <li
      className="ResultsList__StyledListItem-sc-1lpwubp-0 blEIsL"
    >
      <span>
        2
        .
      </span>
      <span>
        Dotaz 1
      </span>
      <span>
        12
        tis.
      </span>
    </li>
    <li
      className="ResultsList__StyledListItem-sc-1lpwubp-0 blEIsL"
    >
      <span>
        3
        .
      </span>
      <span>
        Dotaz 1
      </span>
      <span>
        12
        tis.
      </span>
    </li>
    <li
      className="ResultsList__StyledListItem-sc-1lpwubp-0 blEIsL"
    >
      <span>
        4
        .
      </span>
      <span>
        Dotaz 1
      </span>
      <span>
        12
        tis.
      </span>
    </li>
    <li
      className="ResultsList__StyledListItem-sc-1lpwubp-0 blEIsL"
    >
      <span>
        5
        .
      </span>
      <span>
        Dotaz 1
      </span>
      <span>
        12
        tis.
      </span>
    </li>
  </ul>,
  <button
    className="ResultsList__StyledButton-sc-1lpwubp-2 bEdYZA"
    onClick={[Function]}
  >
    Zobrazit další
  </button>,
]
`;
